src/main.cpp

```diff
@@ -2405,7 +2417,7 @@ void static UpdateTip(CBlockIndex *pindexNew) {
 }
 
 /** Disconnect chainActive's tip. */
-bool static DisconnectTip(CValidationState &state) {
+bool static DisconnectTip(CValidationState &state, bool fBare = false) {
```

Comment:
The name ``fBare`` isn't very clear as to its meaning.

Reply:
Up to here in my review.

Reply:
Upstream replaces it later with a pointer called `disconnectpool`, when they refactored to separate things out of `main.cpp`.

Reply:
OK, but in the meantime I don't know what it means :-)

Reply:
It's set if the mempool should be left unchanged, ie. if the mempool is already bare and should be left bare, such as with `RewindBlockIndex()`.

Reply:
In upstream's later refactor, the pointer `disconnectpool` is set to `nullptr` in this case.